tests/kafkatest/tests/core/replication_test.py

```diff
@@ -128,7 +128,10 @@ def min_cluster_size(self):
     @matrix(failure_mode=["clean_shutdown", "hard_shutdown", "clean_bounce", "hard_bounce"],
             broker_type=["controller"],
             security_protocol=["PLAINTEXT", "SASL_SSL"])
-    def test_replication_with_broker_failure(self, failure_mode, security_protocol, broker_type):
+    @matrix(failure_mode=["clean_shutdown", "hard_bounce"],
+            broker_type=["leader"],
+            security_protocol=["SASL_SSL"], client_sasl_mechanism=["PLAIN"], interbroker_sasl_mechanism=["PLAIN", "GSSAPI"])
```

Comment:
Because these tests take a while to run, maybe we should just have the multiple mechanisms case. What do you think @ewencp?